controllers/metal3.io/host_state_machine.go

```diff
@@ -223,11 +224,10 @@ func (hsm *hostStateMachine) checkInitiateDelete() bool {
 	default:
 		hsm.NextState = metal3v1alpha1.StateDeleting
 	case metal3v1alpha1.StateProvisioning, metal3v1alpha1.StateProvisioned:
-		if len(hsm.Host.Finalizers) > 1 {
-			// Allow other finalizers to run before changing state
-			return false
-		}
 		if hsm.Host.OperationalStatus() == metal3v1alpha1.OperationalStatusDetached {
+			if delayDeleteForDetachedHost(hsm.Host) {
+				return false
```

Comment:
We need a log here.
Prometheus counter wouldn't hurt either.

Reply:
Log is no problem. Are you thinking of a new counter or would an existing one apply to this situation?

Reply:
New counter

Reply:
Done